elastic/kibana
# [Maps] show "no results" found for vector tile aggregations when there are no results

"no results" legend icon is not displayed for heatmap layers and cluster layers (with vector tiles) when there are no matches. Instead, the layer icon is displayed. This is inconsistent with geojson cluster layers, which do show "no results" legend icon when there are no matches. PR also refactors code for extracting values from meta features to put all logic into a single place so its easier to find and test.

To test
1) install web log sample data set
2) create map with heatmap layer or cluster layer with hexagons.
3) add filter to remove all data set. Such as "bytes < 0"
4) legend icon should display "no results" icon 
<img width="500" alt="Screen Shot 2022-04-21 at 4 26 02 PM" src="https://user-images.githubusercontent.com/373691/164561603-8cff8ef4-feb6-4366-a137-e322a4aa0510.png">
